Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.cpp

```diff
@@ -138,6 +177,64 @@ PlatformWheelEvent RemoteLayerTreeEventDispatcher::filteredWheelEvent(const Plat
     return filteredEvent;
 }
 
+DisplayLink* RemoteLayerTreeEventDispatcher::displayLink() const
+{
+    ASSERT(isMainRunLoop());
+    
+    if (!m_scrollingCoordinator)
+        return nullptr;
+
+    auto* drawingArea = dynamicDowncast<RemoteLayerTreeDrawingAreaProxy>(m_scrollingCoordinator->webPageProxy().drawingArea());
+    ASSERT(drawingArea && drawingArea->isRemoteLayerTreeDrawingAreaProxyMac());
```

Comment:
Are you sure you want both of these to be checked by the same line? (The 2 different issues it's checking would show up identically on a stack trace)

Reply:
The conditions are simple enough that I think this is fine (if you hit it, it will be obvious which condition fails).